client/src/lib/dependency-injection/inject.js

```diff
@@ -1,11 +1,14 @@
 import React from 'react';
 import injectorContext from './injectorContext';
 
-const inject = (Component, dependencies, mapDependenciesToProps) => {
+const inject = (dependencies, mapDependenciesToProps, contextArg) => (Component) => {
   // eslint-disable-next-line react/prefer-stateless-function
   class Injected extends React.Component {
     render() {
       let props = {};
+      const context = typeof contextArg === 'function' ?
```

Comment:
I don't like mixed params, it convolutes the API for the sake of reducing a couple of characters typed out for the end-user...

Please enforce a type rather than open for mixed types, at least for the new param that is added, dependencies could be left as is because it's an existing API... although would be nice to get that standardised to a single type as well.